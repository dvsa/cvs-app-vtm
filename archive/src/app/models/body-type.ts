export interface BodyType {
  code: string;
  description: string;
}

export interface IBody {
  make: string;
  model: string;
  functionCode: string;
  conversionRefNo: string;
  bodyType: BodyType;
}

export interface BodyPSV extends Body {
  chassisMake: string;
  chassisModel: string;
}
