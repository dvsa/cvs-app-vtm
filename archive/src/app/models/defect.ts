interface Location {
  axleNumber?: number;
  horizontal?: string;
  vertical: string;
  longitudinal?: string;
  rowNumber: number;
  lateral: string;
  seatNumber: number;
}

interface AdditionalInformation {
  location: Location;
  notes: string;
}

export interface Defect {
  prohibitionIssued: boolean;
  additionalInformation: AdditionalInformation;
  itemNumber: number;
  deficiencyRef: string;
  stdForProhibition: boolean;
  deficiencyId: string;
  imNumber: number;
  deficiencyCategory: string;
  deficiencyText: string;
  prs: boolean;
  deficiencySubId?: any;
  imDescription: string;
  itemDescription: string;
}
