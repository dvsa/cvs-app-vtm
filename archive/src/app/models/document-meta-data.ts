export interface DocumentMetaData {
  uuid: string;
  fileName: string;
  metaName: string;
}

export interface DocumentInfo {
  metaName: string;
  file?: File;
}
