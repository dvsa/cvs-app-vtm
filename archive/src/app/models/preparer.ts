export interface Preparer {
  preparerId: string;
  preparerName: string;
}
