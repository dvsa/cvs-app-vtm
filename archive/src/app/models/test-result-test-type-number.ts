import { TestResultModel } from '@app/models/test-result.model';

export interface TestResultTestTypeNumber {
  testResultUpdated: TestResultModel;
  testTypeNumber: string;
  testResultsUpdated: TestResultModel[];
}
