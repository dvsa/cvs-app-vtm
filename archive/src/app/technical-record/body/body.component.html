<ng-container *ngIf="!editState">
  <dl class="govuk-summary-list ">
    <!-- Make and model have different keys for psv -->
    <div
      class="govuk-summary-list__row"
      *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'trl'"
    >
      <dt class="govuk-summary-list__key">
        Make
      </dt>
      <dd id="test-make" class="govuk-summary-list__value">
        {{ activeRecord.make | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>
    <div
      class="govuk-summary-list__row"
      *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'trl'"
    >
      <dt class="govuk-summary-list__key">
        Model
      </dt>
      <dd id="test-model" class="govuk-summary-list__value">
        {{ activeRecord.model | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">
        Body type
      </dt>
      <dd id="test-bodyTypeDescription" class="govuk-summary-list__value">
        {{ activeRecord.bodyType?.description | DefaultNullOrEmpty | titlecase }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">
        Function
      </dt>
      <dd id="test-functionCode" class="govuk-summary-list__value">
        {{ activeRecord.functionCode | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">
        Conversion reference number
      </dt>
      <dd id="test-conversionRefNo" class="govuk-summary-list__value">
        {{ activeRecord.conversionRefNo | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>
  </dl>

  <ng-container *ngIf="activeRecord.vehicleType === 'psv'">
    <vtm-body-psv [activeRecord]="activeRecord"></vtm-body-psv>
  </ng-container>
</ng-container>

<ng-container *ngIf="editState">
  <dl class="govuk-summary-list">
    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">
        <vtm-body-edit [bodyDetails]="body"></vtm-body-edit>
      </dt>
    </div>
  </dl>
</ng-container>
