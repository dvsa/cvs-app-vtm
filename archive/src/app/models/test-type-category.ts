import { TestCode } from '@app/models/test-code';

export interface TestTypeCategory {
  id: string;
  linkedIds: string[];
  name: string;
  testTypeName: string;
  forVehicleType: string[];
  forVehicleSize: string[];
  forVehicleConfiguration: string[];
  forVehicleAxles: number[];
  forEuVehicleCategory: string[];
  forVehicleClass: string[];
  forVehicleSubclass: string[];
  forVehicleWheels: number[];
  testCodes?: TestCode[];
  nextTestTypesOrCategories?: TestTypeCategory[];
}
