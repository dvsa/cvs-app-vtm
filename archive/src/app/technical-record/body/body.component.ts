import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';

import { TechRecord } from '@app/models/tech-record.model';
import { IBody } from '@app/models/body-type';

@Component({
  selector: 'vtm-body',
  templateUrl: './body.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class BodyComponent implements OnInit {
  @Input() editState: boolean;
  @Input() activeRecord: TechRecord;

  body: IBody;

  constructor() {}

  ngOnInit() {
    const { make, model, functionCode, conversionRefNo, bodyType } = this.activeRecord;

    this.body = {
      make,
      model,
      functionCode,
      conversionRefNo,
      bodyType
    };
  }
}
