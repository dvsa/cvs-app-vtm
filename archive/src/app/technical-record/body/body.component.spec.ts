import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CUSTOM_ELEMENTS_SCHEMA, Component, Input } from '@angular/core';

import { SharedModule } from '@app/shared/shared.module';
import { TESTING_UTILS } from '@app/utils/';
import { IBody } from '@app/models/body-type';
import { BodyComponent } from './body.component';

describe('BodyComponent', () => {
  let component: BodyComponent;
  let fixture: ComponentFixture<BodyComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [BodyComponent, TestBodyEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();

    fixture = TestBed.createComponent(BodyComponent);
    component = fixture.componentInstance;
    component.activeRecord = TESTING_UTILS.mockTechRecord({
      make: 'Marc',
      functionCode: 'a',
      model: 'honda',
      conversionRefNo: 'x239f',
      bodyType: TESTING_UTILS.mockBodyType()
    });
  });

  it('should create view only with populated data', () => {
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });

  it('should render the edit components when editState is true', () => {
    component.editState = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  selector: 'vtm-body-edit',
  template: `
    <div>BodyDetails is: {{ bodyDetails | json }}</div>
  `
})
class TestBodyEditComponent {
  @Input() bodyDetails: IBody;
}
