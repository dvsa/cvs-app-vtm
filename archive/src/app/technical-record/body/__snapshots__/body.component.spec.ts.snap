// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyComponent should create view only with populated data 1`] = `
<vtm-body
  activeRecord={[Function Object]}
  body={[Function Object]}
>
  <dl
    class="govuk-summary-list "
  >
    
    
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Body type 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-bodyTypeDescription"
      >
         Articulated 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Function 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-functionCode"
      >
         a 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Conversion reference number 
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-conversionRefNo"
      >
         x239f 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
  </dl>
</vtm-body>
`;

exports[`BodyComponent should render the edit components when editState is true 1`] = `
<vtm-body
  activeRecord={[Function Object]}
  body={[Function Object]}
  editState={[Function Boolean]}
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        <vtm-body-edit
          ng-reflect-body-details="[object Object]"
        >
          <div>
            BodyDetails is: {
    "make": "Marc",
    "model": "honda",
    "functionCode": "a",
    "conversionRefNo": "x239f",
    "bodyType": {
      "code": "a",
      "description": "articulated"
    }
  }
          </div>
        </vtm-body-edit>
      </dt>
    </div>
  </dl>
</vtm-body>
`;
